File: Builder/python-source/builders/BurgerBuilder.py
from abc import ABC, abstractmethod

from base.Burger import Burger


class BurgerBuilder(ABC):

    @abstractmethod
    def reset(self):
        pass

    @abstractmethod
    def add_low_bread(self):
        pass

    @abstractmethod
    def add_meat(self):
        pass

    @abstractmethod
    def add_high_bread(self):
        pass

    @abstractmethod
    def add_vegetable(self):
        pass

    @abstractmethod
    def get_burger(self) -> Burger:
        pass

File: JavaSource/src/main/java/ru/akhcheck/patterns/builder/builders/BurgerBuilder.java
package ru.akhcheck.patterns.builder.builders;

import ru.akhcheck.patterns.builder.base.Burger;

public interface BurgerBuilder {
    public BurgerBuilder reset();

    public BurgerBuilder addLowBread();
    public BurgerBuilder addMeat();
    public BurgerBuilder addHighBread();
    public BurgerBuilder addVegetable();

    public Burger getBurger();

}
